pkg/operator/periodicbackupcontroller/periodicbackupcontroller.go

```diff
@@ -67,19 +76,35 @@ func NewPeriodicBackupController(
 }
 
 func (c *PeriodicBackupController) sync(ctx context.Context, _ factory.SyncContext) error {
+	klog.Infof("@Mustafa : PeriodicBackupController - sync() begin")
 	if enabled, err := backuphelpers.AutoBackupFeatureGateEnabled(c.featureGateAccessor); !enabled {
 		if err != nil {
 			klog.V(4).Infof("PeriodicBackupController error while checking feature flags: %v", err)
 		}
 		return nil
 	}
 
+	klog.Infof("@Mustafa : PeriodicBackupController - sync() - featuregate enabled")
 	cronJobsClient := c.kubeClient.BatchV1().CronJobs(operatorclient.TargetNamespace)
 	backups, err := c.backupsClient.Backups().List(ctx, v1.ListOptions{})
 	if err != nil {
 		return fmt.Errorf("PeriodicBackupController could not list backup CRDs, error was: %w", err)
 	}
 
+	if len(backups.Items) == 0 && !c.isDefaultBackup {
```

Comment:
if there's no backup, there can't be any default backup by definition, why would you need to invoke the backupVarGetter with this?

Reply:
so if no backup CRs, and the sideCar is not running, then we turn it on with the default specs inlined below